ttnn/cpp/ttnn/tensor/tensor.cpp

```diff
@@ -430,26 +430,20 @@ Tensor Tensor::to(Device* target_device, const MemoryConfig& mem_config) const {
 
 Tensor Tensor::to(DeviceMesh* device_mesh, const MemoryConfig& mem_config) const {
     ZoneScoped;
-    return this->to(device_mesh->get_devices(), mem_config);
+    std::vector<Device*> workers_to_use = distribute_tensor_to_mesh(*this, *device_mesh);
```

Comment:
@dmakoviichuk-tt note this

Reply:
and @tt-asaigal

Reply:
Is it still multidevice only?

Reply:
```
Tensor Tensor::to(DeviceMesh* device_mesh, const MemoryConfig& mem_config) const;
```
Yeah, this is still `MultiDeviceHostStorage` -> `MultiDeviceStorage` what we reviewed last week. `MultiDeviceStorage -> MultiDeviceStorage` will be idempotent. We'll need to revisit this though, thanks for the reminder. We also need to migrate this into under as a registered op :)

Reply:
https://github.com/tenstorrent/tt-metal/issues/11232